common/src/stack/pylib/stack/probepal/main.py

```diff
@@ -0,0 +1,60 @@
+import sys
+from operator import attrgetter
+import importlib
+import pkgutil
+
+from common import UnrecognizedPallet
+
+# general:
+# this package allows for fingerprinting a directory to determine if it contains a stacki pallet
+# the design allows for the dynamic discovery of additional finderprinting probes,
+# along with a means of weighting which probes are used first as well as catching partial matches
+# probes can be added by creating a new python module whose name begins with `prober_` and which
+# has a subclass of the Probe class.  See common.py for more detail. 
+
+
+if __name__ == '__main__':
+	# find and import all 'prober' modules in the directory
+	plugins = {
+		name: importlib.import_module(name)
+		for finder, name, ispkg
+		in pkgutil.iter_modules()
```

Comment:
FYI, apparently [pkgutil.iter_modules](https://docs.python.org/3/library/pkgutil.html#pkgutil.iter_modules) will iterate through all top-level modules on `sys.path` if not passed a path to search.

I'm not sure that this will work properly in the real implementation unless we pass a search path to `pkgutil.iter_modules`, or the program entrypoint (`__main__`) is somehow in the same folder as where the prober modules live (which is my guess as to how the other prober modules are on the top level `sys.path` in this case.)

Reply:
Yeah, I forgot that part.

that line becomes:

`in pkgutil.iter_modules([pathlib.Path(common.__file__).parent])`